### ISSUE TITLE
freebayes Error: signal 11:

### ISSUE THREAD

**Issue by JoonsikJang-NGeneBio** [ekg/freebayes#390]
/workflow-app/freebayes/bin/freebayes[0x4a64a5]
/lib/x86_64-linux-gnu/libc.so.6(+0x36cb0)[0x7ffff6df1cb0]
/workflow-app/freebayes/bin/freebayes[0x45a3b8]
/workflow-app/freebayes/bin/freebayes[0x4600d6]
/workflow-app/freebayes/bin/freebayes[0x4536e5]
/workflow-app/freebayes/bin/freebayes[0x453c0c]
/workflow-app/freebayes/bin/freebayes[0x408642]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5)[0x7ffff6ddcf45]
/workflow-app/freebayes/bin/freebayes[0x40c0c8]

gdb report

(gdb) r -t /workflow_out/39/assay_reference/FFPE48_S3_L001_target_region.bed -I -X --fasta-reference /workflow-dependencies/hg19.fa --min-base-quality 20 --min-alternate-fraction 0.02 /workflow_out/39/data/basecall/alignment/FFPE48_S3_L001_final.bam > FFPE48_S3_L001.unsorted.raw.vcf
Starting program: /workflow-app/freebayes/bin/freebayes -t /workflow_out/39/assay_reference/FFPE48_S3_L001_target_region.bed -I -X --fasta-reference /workflow-dependencies/hg19.fa --min-base-quality 20 --min-alternate-fraction 0.02 /workflow_out/39/data/basecall/alignment/FFPE48_S3_L001_final.bam > FFPE48_S3_L001.unsorted.raw.vcf
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".

Program received signal SIGSEGV, Segmentation fault.
_int_malloc (av=0x7ffff717c760 <main_arena>, bytes=8192) at malloc.c:3740
3740    malloc.c: No such file or directory.
(gdb)
(gdb) bt
#0  _int_malloc (av=0x7ffff717c760 <main_arena>, bytes=8192) at malloc.c:3740
#1  0x00007ffff6e3d6d0 in __GI___libc_malloc (bytes=8192) at malloc.c:2891
#2  0x00007ffff76fddad in operator new(unsigned long) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#3  0x000000000043c326 in allocate (this=0xe36ae8, __n=<optimized out>) at /usr/include/c++/4.8/ext/new_allocator.h:104
#4  _M_allocate (this=0xe36ae8, __n=<optimized out>) at /usr/include/c++/4.8/bits/stl_vector.h:168
#5  std::vector<Allele*, std::allocator<Allele*> >::_M_insert_aux (this=0xe36ae8, __position=, __x=@0x1398f90: 0x10cacf0) at /usr/include/c++/4.8/bits/vector.tcc:345
#6  0x000000000048e76f in std::vector<Allele*, std::allocator<Allele*> >::push_back (this=<optimized out>, __x=<optimized out>) at /usr/include/c++/4.8/bits/stl_vector.h:913
#7  0x0000000000483d10 in getAlleles (haplotypeLength=1, getAllAllelesInHaplotype=false, ignoreProcessedFlag=true, allowedAlleleTypes=114, samples=..., this=0x87c5e0) at AlleleParser.cpp:3690
#8  AlleleParser::getNextAlleles (this=this@entry=0x87c5e0, samples=..., allowedAlleleTypes=allowedAlleleTypes@entry=114) at AlleleParser.cpp:3631
#9  0x00000000004366f0 in main (argc=<optimized out>, argv=<optimized out>) at freebayes.cpp:133

**Comment by JoonsikJang-NGeneBio** [comment]
I found very strange error condition that the folder name affects the error occurrence.

Below command OK
 /NGENEBIO/abcdefghi/freebayes/bin/freebayes -t /NGENEBIO/workflow_out/39/assay_reference/FFPE48_S3_L001_target_region.bed -I -X --fasta-reference /NGENEBIO/workflow-dependencies/hg19.fa --min-base-quality 20 --min-alternate-fraction 0.02 /NGENEBIO/workflow_out/39/data/basecall/alignment/FFPE48_S3_L001_final.bam > test.vcf

Bug Below command Error
 /NGENEBIO/abcde-fghi/freebayes/bin/freebayes -t /NGENEBIO/workflow_out/39/assay_reference/FFPE48_S3_L001_target_region.bed -I -X --fasta-reference /NGENEBIO/workflow-dependencies/hg19.fa --min-base-quality 20 --min-alternate-fraction 0.02 /NGENEBIO/workflow_out/39/data/basecall/alignment/FFPE48_S3_L001_final.bam > test.vcf
Error: signal 11:
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x4a64a5]
/lib/x86_64-linux-gnu/libc.so.6(+0x36cb0)[0x7f35a557acb0]
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x45a3b8]
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x4600d6]
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x4536e5]
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x453c0c]
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x408642]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf5)[0x7f35a5565f45]
/NGENEBIO/abcde-fghi/freebayes/bin/freebayes[0x40c0c8]

**Comment by charlesdavid** [comment]
This issue is preventing me from using FreeBayes. Using version v1.1.0, and this command:
`freebayes -f ${GENOME} -b ${BAM} \
                 --max-complex-gap 50 \
                 -m 10 \
                 --min-coverage 10 \
                 -q 10 | \
                 bcftools convert -O b --threads 2 -o ${OUT}/raw_variants.bcf`

`Error: signal 11:
freebayes[0x4a51f5]
/lib64/libc.so.6(+0x35250)[0x2b9e23f89250]
freebayes[0x42fbe3]
freebayes[0x45365e]
freebayes[0x408322]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x2b9e23f75b35]
freebayes[0x40bda9]`

The BAM was produced and verified using Picard Tools.
It would be helpful to have more informative error messages...
Also has there been any progress on this issue (quiet ~2 months)?
Time is a factor so if no workaround, then will have to move on.

**Comment by ekg** [comment]
This indicates a segmentation fault. Library ABI incompatibility is the
reason I most recently saw it. Can you confirm which commit this occurs on?

**Comment by IgnasiLucas** [comment]
Hello. I am having the same issue. I don't see obvious errors in the bam file, and I keep getting the error signal 11. I am using the most recent commit, version:  v1.1.0-44-gd784cf8.

**Comment by ekg** [comment]
You built all the way from a "fresh" clone?

**Comment by IgnasiLucas** [comment]
Yes, I did.

**Comment by ekg** [comment]
I wonder why this isn't showing up in automated testing. Is your
environment clean? Do you have things like htslib installed? If so, what
versions? Also, what gcc version are you using?

**Comment by IgnasiLucas** [comment]
Yes, I have htslib installed, version 1.5-5-gda5c0c7.
And the gcc I have is version 4.9.2 (Debian 4.9.2-10).

**Comment by chambm** [comment]
I have the same problem with some of my BAMs. It seems to happen mostly with lower quality BAMs:

I built freebayes with `make debug` and ran `gdb`:
```
freebayes --version
 v1.1.0-44-gd784cf8-dirty
```

```
(gdb) Starting program: /vol_b/jobs/004/4922/working2/freebayes/bin/freebayes --bam b1.sam --fasta-reference /galaxy/data/hg38/sam_index/hg38.fa --vcf ./vcf_output/part_chrUn_GL000216v2:0..176608.vcf -O
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/lib/x86_64-linux-gnu/libthread_db.so.1".
*** Error in `/vol_b/jobs/004/4922/working2/freebayes/bin/freebayes': double free or corruption (!prev): 0x0000000004b503b0 ***

Program received signal SIGABRT, Aborted.
0x00007ffff69bfc37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
56      ../nptl/sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) #0  0x00007ffff69bfc37 in __GI_raise (sig=sig@entry=6) at ../nptl/sysdeps/unix/sysv/linux/raise.c:56
#1  0x00007ffff69c3028 in __GI_abort () at abort.c:89
#2  0x00007ffff69fc2a4 in __libc_message (do_abort=do_abort@entry=1, fmt=fmt@entry=0x7ffff6b0def0 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff6a0856e in malloc_printerr (ptr=<optimized out>, str=0x7ffff6b0dfd8 "double free or corruption (!prev)", action=1) at malloc.c:4996
#4  _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3840
#5  0x00007ffff7328529 in std::string::assign(std::string const&) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6
#6  0x00000000005819e8 in Allele::update (this=0xf52230, haplotypeLength=1) at Allele.cpp:31
#7  0x00000000005b4fa1 in AlleleParser::getAlleles (this=0xa255e0, samples=..., allowedAlleleTypes=126, haplotypeLength=1, getAllAllelesInHaplotype=false, ignoreProcessedFlag=true) at AlleleParser.cpp:3613
#8  0x00000000005b4bb3 in AlleleParser::getNextAlleles (this=0xa255e0, samples=..., allowedAlleleTypes=126) at AlleleParser.cpp:3557
#9  0x00000000005509f4 in main (argc=8, argv=0x7fffffffe238) at freebayes.cpp:135
```

```
gcc --version
gcc (Ubuntu 4.8.4-2ubuntu1~14.04.3) 4.8.4
```

Here's the SAM file:
[b1.zip](https://github.com/ekg/freebayes/files/1153268/b1.zip)

**Comment by ekg** [comment]
@arrigon using your data I am not able to reproduce this failure as of the current HEAD of master. Can you check to see if this is resolved for you and if so please close it? It may have been resolved last month with 8d2b3a06.

**Comment by hzpc-joostk** [comment]
Hello! We get a similar error with version 1.3.2 (dirty). 

TL;DR. In our case, the problem seems to be an empty `--haplotype-basis-alleles` file. 

```sh
$ freebayes --version
version:  v1.3.2-dirty

conda list freebayes
# Name                    Version                   Build  Channel
freebayes                 1.3.2            py37hc088bd4_0    bioconda
```
From an experiment with 2736 different BAM files, 522 BAM files raised the same error (on stderr):
```
Error: signal 11:
freebayes(+0xbd19b)[0x55555561119b]
/lib64/libc.so.6(+0x363b0)[0x2aaaab4493b0]
freebayes(+0xbddcb)[0x555555611dcb]
freebayes(+0x6dae6)[0x5555555c1ae6]
freebayes(+0x53fae)[0x5555555a7fae]
freebayes(+0x68f23)[0x5555555bcf23]
freebayes(+0xa1cb)[0x55555555e1cb]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x2aaaab435505]
freebayes(+0x10ea2)[0x555555564ea2]
```
Nothing was printed on stdout, not even a VCF header.

We run a script where we launch FreeBayes twice, first with `--haplotype-length -1` (along with some common options, e.g. `--fasta-reference`), then do some filtering on its output, `bgziptabix` the filtered output and then run FreeBayes again with `--haplotype-length 200 --haplotype-basis-alleles filtered.vcf.gz`.

The second FreeBayes always fails when `filtered.vcf.gz` is empty. In case of these 522 BAM files, the coverage was too low to identify any variants.

I tested a BAM file that worked, but mocked an empty VCF to be fed to FreeBayes in the second run. Then it fails with the same error. Enough coverage in the BAM, but no variants as haplotype basis alleles.

**Comment by neelam19051** [comment]
Hi, i am getting same error-

freebayes --version
version:  v1.3.6

 gcc --version
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

The error are-

Error: signal 11:
freebayes(+0x8f3fd)[0x5591633853fd]
/lib/x86_64-linux-gnu/libc.so.6(+0x43090)[0x7f3a7638b090]
/home/bvs/anaconda3/envs/myenv/bin/../lib/libstdc++.so.6(+0xd7ce7)[0x7f3a767b6ce7]
/home/bvs/anaconda3/envs/myenv/bin/../lib/libstdc++.so.6(_ZNKSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE17find_first_not_ofEPKcmm+0x33)[0x7f3a767e6913]
freebayes(+0x9044d)[0x55916338644d]
freebayes(+0x50e74)[0x559163346e74]
freebayes(+0x39bb5)[0x55916332fbb5]
freebayes(+0x3ea09)[0x559163334a09]
freebayes(+0xd956)[0x559163303956]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf3)[0x7f3a7636c083]


Thank you!

**Comment by j-a-thia** [comment]
Also with v1.3.6 from recent install through conda:

Error: signal 11:
freebayes(+0x90cdb)[0x5555555e4cdb]
/lib64/libc.so.6(+0x36280)[0x2aaaab406280]
/nobackup/beegfs/workspace/jt829/miniconda3/envs/freebayes_env/bin/../lib/libstdc++.so.6(_ZNKSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEE17find_first_not_ofEPKcmm+0x2b)[0x2aaaaaddc7ef]
freebayes(+0x91d4d)[0x5555555e5d4d]
freebayes(+0x52eea)[0x5555555a6eea]
freebayes(+0x38837)[0x55555558c837]
freebayes(+0x3d4b8)[0x5555555914b8]
freebayes(+0xd9db)[0x5555555619db]
/lib64/libc.so.6(__libc_start_main+0xf5)[0x2aaaab3f23d5]
freebayes(+0x12ea5)[0x555555566ea5]

**Comment by pjotrp** [comment]
If you share the input file(s) and command line I could have a go.  Alternatively build the debug version of freebayes and show that stack trace.

**Comment by Reba13** [comment]
Could someone explain, please, what is causing this error?
freebayes v1.3.7
used: "freebayes -f hs37d5.fa test.bam >var.vcf"

Error: signal 11:
0   freebayes                           0x0000000104c2f340 _Z15segfaultHandleri + 32
1   libsystem_platform.dylib            0x00007ff816fbe37d _sigtramp + 29
2   ???                                 0x0000000104dd6080 0x0 + 4376584320
3   freebayes                           0x0000000104bfd6de _Z20allPossibleGenotypesiRNSt3__16vectorI6AlleleNS_9allocatorIS1_EEEE + 46
4   freebayes                           0x0000000104c0224f _Z20getGenotypesByPloidyRNSt3__16vectorIiNS_9allocatorIiEEEERNS0_I6AlleleNS1_IS5_EEEE + 129
5   freebayes                           0x0000000104bb92e1 main + 6705
6   dyld                                0x00007ff816c083a6 start + 1942
